src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/TreeViewTests.cs

```diff
@@ -0,0 +1,795 @@
+			};
+		}
+
+		[TestMethod]
+		public void TreeViewItemsSourceTest()
+		{
+			var treeView = new TreeView();
+			var items = CreateTreeViewItemsSource();
+			treeView.ItemsSource = items;
+
+			Verify.AreEqual(treeView.RootNodes.Count, 2);
+			Verify.AreEqual(treeView.RootNodes[0].Content as TreeViewItemSource, items[0]);
+		}
+
+		[TestMethod]
+		[RunsOnUIThread]
+		public void TreeViewItemsSourceUpdateTest()
+		{
+			var treeView = new TreeView();
+			var items = CreateTreeViewItemsSource();
+			treeView.ItemsSource = items;
+
+			// Insert
+			var newItem = new TreeViewItemSource() { Content = "newItem" };
+			items.Add(newItem);
+			Verify.AreEqual(treeView.RootNodes.Count, 3);
+			var itemFromNode = treeView.RootNodes[2].Content as TreeViewItemSource;
+			Verify.AreEqual(newItem.Content, itemFromNode.Content);
+
+			// Remove
+			items.Remove(newItem);
+			Verify.AreEqual(treeView.RootNodes.Count, 2);
+
+			// Replace
+			var item3 = new TreeViewItemSource() { Content = "3" };
+			items[1] = item3;
+			itemFromNode = treeView.RootNodes[1].Content as TreeViewItemSource;
+			Verify.AreEqual(item3.Content, itemFromNode.Content);
+
+			// Clear
+			items.Clear();
+			Verify.AreEqual(treeView.RootNodes.Count, 0);
+		}
+
+		[TestMethod]
+		[RunsOnUIThread]
+		public void TreeViewNodeStringableTest()
+		{
+			var node = new TreeViewNode() { Content = "Node" };
+			Verify.AreEqual(node.Content, node.ToString());
+
+			// Test inherited type
+			var node2 = new TreeViewNode2() { Content = "Inherited from TreeViewNode" };
+			Verify.AreEqual(node2.Content, node2.ToString());
+		}
+
+		[TestMethod]
+		public async Task TreeViewPendingSelectedNodesTest()
+		{
+			TreeView treeView = null;
+			TreeViewNode node1 = null;
+			TreeViewNode node2 = null;
+			await RunOnUIThread.Execute(() =>
+			{
+				treeView = new TreeView();
+				treeView.SelectionMode = TreeViewSelectionMode.Multiple;
+
+				node1 = new TreeViewNode() { Content = "Node1" };
+				node2 = new TreeViewNode() { Content = "Node2" };
+				treeView.RootNodes.Add(node1);
+				treeView.RootNodes.Add(node2);
+				treeView.SelectedNodes.Add(node1);
+
+				TestServices.WindowHelper.WindowContent = treeView;
+			});
+
+			await TestServices.WindowHelper.WaitForIdle();
+
+			await RunOnUIThread.Execute(() =>
+			{
+				Verify.AreEqual(true, IsMultiSelectCheckBoxChecked(treeView, node1));
+				Verify.AreEqual(false, IsMultiSelectCheckBoxChecked(treeView, node2));
+			});
+		}
+
+		////TODO:
+		////[TestMethod]
+		////public void VerifyVisualTree()
+		////{
+		////	TreeView treeView = null;
+
+		////	treeView = new TreeView() { Width = 400, Height = 400 };
+		////	var node1 = new TreeViewNode() { Content = "Node1" };
+		////	treeView.RootNodes.Add(node1);
+		////	TestUtilities.SetAsVisualTreeRoot(treeView);
+
+		////	VisualTreeTestHelper.VerifyVisualTree(root: treeView, verificationFileNamePrefix: "TreeView");
+		////}
+
+		//[TestMethod]
+		//public void TreeViewSelectionChangedSingleMode()
+		//{
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)